cdap-client-tests/src/test/java/co/cask/cdap/client/PreferencesClientTestRun.java

```diff
@@ -83,6 +84,8 @@ public void testProgramAPI() throws Exception {
                                                         AppReturnsArgs.SERVICE));
       programClient.start(AppReturnsArgs.NAME, ProgramType.SERVICE, AppReturnsArgs.SERVICE,
                           ImmutableMap.of("run", "value"));
```

Comment:
The difference is that in the `ServiceClient` we simply return the constructed URL instead of actually doing service discovery (since the client is external and has no way to do discovery). Maybe the right thing to do would be for `start` calls to actually wait for the Service/Flow to startup.